template.tpl

```diff
@@ -96,12 +96,156 @@ ___TEMPLATE_PARAMETERS___
     "name": "eventType",
     "type": "SELECT"
   },
+  {
+    "type": "GROUP",
+    "name": "eventMetadataGroup",
+    "displayName": "Additional Event Information",
+    "groupStyle": "NO_ZIPPY",
+    "subParams": [
+      {
+        "type": "SIMPLE_TABLE",
+        "name": "eventMetadataParams",
+        "displayName": "",
+        "simpleTableColumns": [
+          {
+            "defaultValue": "",
+            "displayName": "Parameter Name",
+            "name": "name",
+            "type": "SELECT",
+            "selectItems": [
+              {
+                "value": "itemCount",
+                "displayValue": "Item Count"
+              },
+              {
+                "value": "value",
+                "displayValue": "Event Value"
+              },
+              {
+                "value": "currency",
+                "displayValue": "Currency"
+              },
+              {
+                "value": "transactionId",
+                "displayValue": "Transaction ID"
+              }
+            ],
+            "isUnique": true,
+            "macrosInSelect": false
+          },
+          {
+            "defaultValue": "",
+            "displayName": "Parameter Value",
+            "name": "value",
+            "type": "TEXT",
```

Comment:
Hmmm.  This gives me the willies.  `value` and `itemCount` are expected to be JavaScript type `'number'`.  They'll be filtered out if they aren't.  Can we do the conversion in `getRdt()` ourselves?

Reply:
Or how else can we do this (different types for the fields)?

Reply:
Btw, the test is good.  Thanks.  But I don't think it effectively tests this case because it itself injects the data it wants/expects (numbers not strings as you made in e573d55).

Reply:
I think this `"type": "TEXT"` is actually referring to the input field type (i.e. `SELECT` for dropdowns, `TEXT` for text input)

Reply:
From Slack:

> Monica Ting
> I think realistically advertisers will not be entering values directly on the GTM tag configuration itself, but we should advise them to utilize Data Layer Variables to pass each specific event’s metadata to the tag(s).
> It seems like the GTM-provided validation rules are for when entering static values on the tag configuration but will ignore validating on any variables it sees. For validating variable values, we will probably need to insert our own JS code in the template, which may not be super maintainable for same reason as the previous comment on currency length. If the advertiser passes a value of type number to one of the data layer variables, it will also pass it as a number to the pixel.
> The only thing I’m not sure is if the advertiser does choose to enter a static value for these fields, it will still send them as strings, even with the GTM-provided validations in place. I guess this is the original question you both posed.
> I’m wondering if it’s worth trying to account for this case or  hopefully we can just make the instructions for utilizing the Data Layer Variables clear.

> Eseosa Asiruwa
> Ah ok, I'm not familiar with GTM so thank you for the clarification! My main concern was about being able to validate that if given value or itemCount, the data given should be a  number. We can make sure instructions are clear about using the Data Layer Variables but, from my understanding, it seems like we'd be leaving this piece of validation to the advertisers.